packages/language-server/server/server.js

```diff
@@ -73,181 +37,27 @@ connection.onHover(meta => {
-    path.join(__dirname, "../node_modules/@hegel/typings/nodejs/globals.d.ts"),
-    globalScope
-  );
-  return nodeJsGlobalTypeGraph;
-}
 
-async function getBrowserTypeDefinitions(globalScope) {
-  browserGlobalTypeGraph = await getTypeGraphFor(
-    path.join(__dirname, "../node_modules/@hegel/typings/browser/index.d.ts"),
-    globalScope
-  );
-  return browserGlobalTypeGraph;
-}
-
-async function getModuleAST(currentModulePath) {
-  if (!config) {
-    config = await getConfig(currentModulePath);
-  }
-  return importModule(config, async (modulePath, isTypings) => {
-    let moduleContent = await readFile(modulePath, "utf8");
-    moduleContent =
-      path.extname(modulePath) === ".json"
-        ? wrapJSON(moduleContent)
-        : moduleContent;
-    const config = isTypings ? dtsrc : babelrc;
-    return babylon.parse(moduleContent, config);
-  }, true);
-}
-
-function wrapJSON(content) {
-  return `export default ${content}`;
-}
-
-function mixTypeDefinitions(config) {
-  return async globalScope => {
-    if (stdLibTypeGraph === undefined) {
-      stdLibTypeGraph = await getStandardTypeDefinitions(globalScope);
-    }
-    mixSomeTypeDefinitions(globalScope, stdLibTypeGraph);
-    const shouldIncludeNodeJS = config.environment.includes("nodejs");
-    const shouldIncludeBrowser = config.environment.includes("browser");
-    const waitingTypes = [
-      shouldIncludeNodeJS && nodeJsGlobalTypeGraph === undefined
-        ? getNodeJSTypeDefinitions(globalScope)
-        : nodeJsGlobalTypeGraph,
-      shouldIncludeBrowser && browserGlobalTypeGraph === undefined
-        ? getBrowserTypeDefinitions(globalScope)
-        : browserGlobalTypeGraph
-    ];
-    const [nodeJsGlobalScope, browserGlobalScope] = await Promise.all(
-      waitingTypes
-    );
-    if (shouldIncludeNodeJS) {
-      mixSomeTypeDefinitions(globalScope, nodeJsGlobalScope);
-    }
-    if (shouldIncludeBrowser) {
-      mixSomeTypeDefinitions(globalScope, browserGlobalScope);
-    }
-  };
-}
-
-function mixSomeTypeDefinitions(globalScope, additionalTypeGraph) {
-  for (const [name, variable] of additionalTypeGraph.body.entries()) {
-    variable.parent = globalScope;
-    globalScope.body.set(name, variable);
-  }
-  for (const [name, type] of additionalTypeGraph.typeScope.body.entries()) {
-    type.parent = globalScope.typeScope;
-    globalScope.typeScope.body.set(name, type);
-  }
-}
-
-function formatErrorRange(error) {
-  const isSyntaxError = error instanceof SyntaxError;
-  return {
-    start: isSyntaxError
-      ? { ...error.loc, line: error.loc.line - 1 }
-      : convertLocToRange(error.loc.start),
-    end: isSyntaxError
-      ? { ...error.loc, column: 1000 }
-      : convertLocToRange(error.loc.end)
-  };
-}
-
-function convertRangeToLoc(loc) {
-  return {
-    line: loc.line + 1,
-    column: loc.character
-  };
-}
+connection.onDidChangeWatchedFiles(async (change) => {
+  const diagnostics = await validateTextDocument(change.document);
```

Comment:
And also, as a result of adding `setTimeout`, we will have duplication of the logic, so, will be cool to move the logic into separated function inside the module.